backend/tests/unit/apps/owasp/admin/contribution_score_test.py

```
@@ -0,0 +1,165 @@
+    def test_recalculate_mixed_success_and_failure(self, mock_calculator_cls):
+        """Test recalculate action with mixed success and failure results."""
+        mock_calculator = MagicMock()
+        mock_calculator_cls.return_value = mock_calculator
+
+        score_ok = Mock()
+        score_ok.github_user.login = "ok_user"
+
+        score_fail = Mock()
+        score_fail.github_user.login = "fail_user"
+
+        def side_effect(user):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `side_effect`

Add return type annotation: `None`

(ANN202)